CI failure: E2E Tests — __dirname undefined in ES modules causing Docker build failures

## CI Failure Report

**Workflow:** E2E Tests
**Branch:** main
**Commit:** d63c70da817b882860d970647735045017e82fbf
**Run:** https://github.com/Action-Llama/action-llama/actions/runs/23395144522

## Failure Summary

E2E tests are failing because `__dirname` is undefined in ES modules, causing Docker build image path resolution to fail when trying to locate Dockerfiles.

## Failed Job(s)

### e2e

**Error output:**
```
⎯⎯⎯⎯⎯ Uncaught Exception ⎯⎯⎯⎯⎯
Error: ENOENT: no such file or directory, lstat '/home/runner/work/action-llama/packages/e2e/docker/local/Dockerfile'
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: { errno: -2, code: 'ENOENT', syscall: 'lstat', path: '/home/runner/work/action-llama/packages/e2e/docker/local/Dockerfile' }
This error originated in "src/tests/cli-flows.test.ts" test file.

##[error]Error: Test timed out in 300000ms.
```

Multiple tests failed with timeouts because the Docker container creation was failing.

## Root Cause Analysis

The issue was introduced in commit d63c70da817b882860d970647735045017e82fbf, which attempted to fix path resolution by replacing `process.cwd()` with `__dirname` in the `buildImage` method of `packages/e2e/src/harness.ts`.

However, this is an ES module project (as evidenced by `.js` imports and the `tsconfig.json` with `"module": "Node16"`), and `__dirname` is not available in ES modules - it's a CommonJS global. When `__dirname` is undefined, the path resolution fails:

```typescript
// Current broken code in harness.ts line ~260:
const repoRoot = path.resolve(__dirname, "../../../.."); // __dirname is undefined!
```

The Dockerfile exists at the correct location, but the path calculation is broken due to the undefined `__dirname`.

## Suggested Fix

Replace the `__dirname` usage with the ES module equivalent. In `packages/e2e/src/harness.ts`, change:

```diff
+ import { fileURLToPath } from 'url';
+ import { dirname } from 'path';

  private async buildImage(imageName: string, contextPath: string) {
-   // Use __dirname to get the absolute path to the harness.ts file, then navigate to repo root
-   const repoRoot = path.resolve(__dirname, "../../../..");
+   // Use ES module equivalent of __dirname to get the absolute path to the harness.ts file
+   const __filename = fileURLToPath(import.meta.url);
+   const __dirname = dirname(__filename);
+   const repoRoot = path.resolve(__dirname, "../../../..");
    const absoluteContextPath = path.resolve(repoRoot, "packages/e2e", contextPath);
    const dockerfilePath = path.join("packages/e2e", contextPath, "Dockerfile");
```

## Additional Context

This issue is part of a series of attempts to fix Docker build context path resolution:
- d63c70d Fix E2E Docker build context path resolution failure (#247) ← **Current failing commit**
- 9794441 Fix E2E Docker build context paths (#245)
- f03f650 fix: correct E2E Docker build context paths (closes #242) (#243)

The commit message indicates this was meant to fix issue #246, but the ES module incompatibility was not caught before merging.

---
*This issue was automatically created by the gh-actions-responder agent.*

--- a/packages/e2e/src/harness.ts
+++ b/packages/e2e/src/harness.ts
@@ -3,6 +3,8 @@ import { randomUUID } from "crypto";
 import { generateKeyPairSync } from "crypto";
 import { promises as fs } from "fs";
 import path from "path";
+import { fileURLToPath } from 'url';
+import { dirname } from 'path';
 import { Client as SSHClient } from "ssh2";
 import { assertDockerAvailable, isDockerAvailable } from "./docker-utils.js";
 
@@ -257,7 +259,9 @@ export class E2ETestContext {
   }
 
   private async buildImage(imageName: string, contextPath: string) {
-    // Use __dirname to get the absolute path to the harness.ts file, then navigate to repo root
+    // Use ES module equivalent of __dirname to get the absolute path to the harness.ts file
+    const __filename = fileURLToPath(import.meta.url);
+    const __dirname = dirname(__filename);
     const repoRoot = path.resolve(__dirname, "../../../..");
     const absoluteContextPath = path.resolve(repoRoot, "packages/e2e", contextPath);
     const dockerfilePath = path.join("packages/e2e", contextPath, "Dockerfile");

